add darwin arm64 build target, improve m1 devx

joefitzgerald opened the pull request:
Signed-off-by: Joe Fitzgerald <jfitzgerald@vmware.com>

## What this PR does / why we need it
<!--
Add detailed explanation of what this PR does and why it is
needed.
-->
This PR causes darwin / arm64 binaries to be generated by `make release`. It also makes improvements to scripts that support building a release locally on an arm64 Mac.

## Details for the Release Notes
<!--
Unless this is a trivial change, we want to know more about your contribution!
This can even be a TLDR version of the "What this PR does".
If a trivial change, just write "NONE" in the release-note block below.
Otherwise, a release note is required:
-->
```release-note
generate arm64 binaries for darwin
```

## Which issue(s) this PR fixes
<!--
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
-->
N/A

## Describe testing done for PR
<!--
Example: Created vSphere workload cluster to verify change. 
-->
* ran `make release` locally and validated it succeeds

## Special notes for your reviewer
<!--
Add any things that reviewers should be aware of as they review
your PR.

Example: Please verify how I handled foo aligns with overall plan.
-->
* This PR will be accompanied with a `cayman_vmware-tanzu_tce` MR to ensure the new binaries are notarized correctly.

davidvonthenen reviewed (commented):
I left a comment... and I'm going to private message you about the signing process.

davidvonthenen reviewed (approved):
This looks good to me!

joefitzgerald committed:
add darwin arm64 build target, improve m1 devx

Signed-off-by: Joe Fitzgerald <jfitzgerald@vmware.com>

Update .github/workflows/release-nonga.yaml

Co-authored-by: Navid Shaikh <shaikhnavid14@gmail.com>

- explicitly set envs for tanzu framework build

Signed-off-by: Joe Fitzgerald <jfitzgerald@vmware.com>

davidvonthenen reviewed (changes requested):
Please see comment above. This PR is good but I have work on the build infra side to support this.

EDIT: Going to try to do this week.

nrb reviewed (commented):
@joefitzgerald Are you able to build the `tanzu` binary with this?

I ask because the tanzu-framework Makefile [defines a small subset of environments](https://github.com/vmware-tanzu/tanzu-framework/blob/main/Makefile#L83), and it's [checked](https://github.com/vmware-tanzu/tanzu-framework/blob/main/Makefile#L248) such that it shouldn't compile.

I did a search for PRs on tanzu-framework but couldn't find any.

davidvonthenen reviewed (approved):
LGTM. thanks for this awesome contribution!
